src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts

```diff
@@ -330,7 +330,13 @@ export class CodeCell extends Disposable {
 			this.templateData.editor?.focus();
 		}
 
+		if (this.notebookEditor.getActiveCell() === this.viewCell
```

Comment:
Should this method do the same for output focus? But also, since `focusNotebookCell` on the editor does `cell.focusMode = CellFocusMode.Container;`, could this lead to infinite recursion?

Reply:
Actually, if this cell is already the active cell, why do you have to call `focusNotebookCell` here?

Reply:
`focusNotebookCell` has quite some side effects (updating selection model, asking list view to focus DOM, etc), calling it is fixing up what's broken due to `focus/blur` native browser event.

Also this should not be here, as it's calling back to the notebook editor. We should put this code in https://github.com/microsoft/vscode/blob/a8c6745e4e7b5a51df2c8910d81b01d0040842af/src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts#L1536-L1538